src/common/json_helper.h

```diff
@@ -493,6 +494,26 @@ inline bool json_decode(const JsonObject &in, std::set<T> &t)
     return true;
 }
 
+template <typename T1, typename T2>
```

Comment:
`typename T2` can be removed?

Reply:
No need, if i want to use custom type data, just add a new hash class<T2>.